apps/voting-disputable/app/src/script.js

```diff
@@ -358,6 +359,75 @@ function loadVoteDisputableInfo(voteId) {
   )
 }
 
+function getAgreement() {
+  return retryEvery(() =>
+    app
+      .call('getAgreement')
+      .toPromise()
+      .catch(err => {
+        console.error(`Error fetching agreement address`, err)
+        throw err
+      })
+  )
+}
+
+async function getDisputableAction(actionId) {
+  const agreementAddress = await getAgreement()
+  const agreement = app.external(agreementAddress, agreementAbi)
+
+  return agreement
+    .getAction(actionId)
+    .toPromise()
+    .then(action => getChallenge(agreement, action))
+    .then(action => getCollaterall(agreement, action))
+    .then(action => getCollaterallToken(action))
+}
+
+async function getChallenge(agreement, action) {
+  if (action.currentChallengeId && parseInt(action.currentChallengeId) > 0) {
+    return {
+      ...action,
+      challenge: await agreement
+        .getChallenge(action.currentChallengeId)
+        .toPromise(),
+    }
+  }
+  return action
+}
+
+async function getCollaterall(agreement, action) {
```

Comment:
I think these function names should probably be updated as `Collaterall` -> `Collateral`